File: PagingRemoteMediatorJava/app/src/main/java/com/websarva/wings/android/pagingremotemediatorjava/data/repository/PhoneRepository.java
package com.websarva.wings.android.pagingremotemediatorjava.data.repository;

import android.app.Application;

import androidx.annotation.OptIn;
import androidx.paging.ExperimentalPagingApi;
import androidx.paging.Pager;
import androidx.paging.PagingConfig;

import com.websarva.wings.android.pagingremotemediatorjava.data.local.AppDatabase;
import com.websarva.wings.android.pagingremotemediatorjava.data.local.Phone;
import com.websarva.wings.android.pagingremotemediatorjava.data.local.PhoneDAO;
import com.websarva.wings.android.pagingremotemediatorjava.data.remote.PhoneRemoteMediator;


public class PhoneRepository {
	private static final int ITEMS_PER_PAGE = 50;
	private AppDatabase _db;

	public PhoneRepository(Application application) {
		_db = AppDatabase.getDatabase(application);
	}

	@OptIn(markerClass = ExperimentalPagingApi.class)
	public Pager<Integer, Phone> getAllPhoneListPager() {
		PhoneDAO phoneDAO = _db.createPhoneDAO();
		PagingConfig pagingConfig = new PagingConfig(ITEMS_PER_PAGE);
		PhoneRemoteMediator phoneRemoteMediator = new PhoneRemoteMediator(_db);
		Pager<Integer, Phone> phoneListPager = new Pager<>(pagingConfig, null, phoneRemoteMediator, () -> phoneDAO.findAll());
		return phoneListPager;
	}
}

File: PagingRemoteMediatorKotlin/app/src/main/java/com/websarva/wings/android/pagingremotemediatorkotlin/data/repository/PhoneRepository.kt
package com.websarva.wings.android.pagingremotemediatorkotlin.data.repository

import android.app.Application
import androidx.paging.ExperimentalPagingApi
import androidx.paging.Pager
import androidx.paging.PagingConfig
import com.websarva.wings.android.pagingremotemediatorkotlin.data.local.AppDatabase
import com.websarva.wings.android.pagingremotemediatorkotlin.data.local.Phone
import com.websarva.wings.android.pagingremotemediatorkotlin.data.remote.PhoneRemoteMediator

private const val ITEMS_PER_PAGE = 50

class PhoneRepository(application: Application) {
	private val _db: AppDatabase

	init {
		_db = AppDatabase.getDatabase(application)
	}

	@OptIn(ExperimentalPagingApi::class)
	fun getAllPhoneListPager(): Pager<Int, Phone> {
		val phoneDAO = _db.createPhoneDAO()
		val pagingConfig = PagingConfig(ITEMS_PER_PAGE)
		val phoneListPager = Pager(pagingConfig, null, PhoneRemoteMediator(_db)) { phoneDAO.findAll() }
		return phoneListPager
	}
}
